test/stdtypes/concurrency_test.go

```diff
@@ -0,0 +1,51 @@
+package stdtypes
+
+import (
+	"io/ioutil"
+	"sync"
+	"testing"
+
+	"github.com/gogo/protobuf/proto"
+)
+
+func TestConcurrentTextMarshal(t *testing.T) {
```

Comment:
Does this test fail when you run it with -test.race ?
Or how does this test fail.

If it requires -test.race then maybe we should make a Makefile that actually calls this with -test.race to make sure we don't revert this change in future.